File: gamerecord.cc
#include "gamerecord.h"
#include <iostream>

using namespace std;

GameRecord::GameRecord(int id, string white, string black, string date): 
	id(id), white(white), black(black), date(date) {}

void GameRecord::printInfo() {
	cout << id << " " << white << " " << black << " " << date << endl;
}

string GameRecord::getWhite() {
	return white;
}

string GameRecord::getBlack() {
	return black;
}

string GameRecord::getDate() {
	return date;
}

File: src/com/skysketches/chess/GameRecord.java
/**
 * Project: Chess
 * Author: Ming-Cee Yee
 * Created: 2015-09-15
 * Description: Stores metadata of save games
 */

package com.skysketches.chess;

public class GameRecord {

	private int id;
	private String white;
	private String black;
	private String date;
	
	public GameRecord(int i, String w, String b, String d) {
		this.id = i;
		this.white = w;
		this.black = b;
		this.date = d;
	}
	
	/**
	 * printInfo() prints the values of the GameRecord fields to standard output
	 */
	public void printInfo() {
		System.out.println(id + " " + white + " " + black + " " + date);
	}
	
	/**
	 * getWhite() returns the name of the white player
	 */
	public String getWhite() {
		return this.white;
	}
	
	/**
	 * getBlack() returns the name of the black player
	 */
	public String getBlack() {
		return this.black;
	}
	
	/**
	 * getDate() returns the date of the game
	 */
	public String getDate() {
		return this.date;
	}

}
